### ISSUE TITLE
derive es mapping from spec

### ISSUE THREAD

**Issue by loomis** [slipstream/SlipStreamServer#1497]
- [ ] Determine how to derive the Elasticsearch mapping from the clojure.spec definition for each CIMI resource.
 - [ ] Implement for existing resources.
 - [ ] Verify that the mapping corresponds to the actual mapping in Elasticsearch for existing resources.

**Comment by elegoff** [comment]
The https://github.com/metosin/spec-tools project allows interesting specs handling

**Comment by loomis** [comment]
This looks promising.  The conversion to a JSON document is probably the most useful example for what we need to do.  The document format for the [Elasticsearch mappings](https://www.elastic.co/guide/en/elasticsearch/reference/current/mapping.html) is similar to the JSON example.

An example based on the docs:
```json
PUT my_index 
{
  "mappings": {
    "_doc": { 
      "properties": { 
        "title":    { "type": "text"  }, 
        "name":     { "type": "text"  }, 
        "age":      { "type": "integer" },  
        "created":  {
          "type":   "date", 
          "format": "strict_date_optional_time||epoch_millis"
        }
      }
    }
  }
}
```
Note that the type should always be "_doc" with ES 6.

I would suggest creating a prototype function that would produce an ES mapping based on the spec-tools JSON type function.  The first version wouldn't need to deal with dates for differences between keyword/text.

**Comment by loomis** [comment]
Included in PR #1500.
